src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4DeviceAndPatientCreateIdentityService.cs

```diff
@@ -9,9 +9,14 @@
 using Hl7.Fhir.Rest;
 using Microsoft.Health.Extensions.Fhir;
 using Microsoft.Health.Extensions.Fhir.Service;
+using Microsoft.Health.Fhir.Ingest.Config;
 using Microsoft.Health.Fhir.Ingest.Data;
+using Microsoft.Health.Fhir.Ingest.Host;
+using Microsoft.Health.Fhir.Ingest.Service;
 using Model = Hl7.Fhir.Model;
 
+[assembly: ResourceIdentityService(ResourceIdentityServiceType.Create, typeof(R4DeviceAndPatientCreateIdentityService))]

```

Comment:
You want to have the attribute on the class not the assembly. If you do that you don't need the type attribute since it is implicit since the attribute is already associated with the desired class.

Reply:
Thanks for asking, I compared two approaches and decide to use the assembly one.

I did some research of how to get all types given an attribute type, and I didn't find anything for that, probably the type doesn't get reverse indexed with the attributes. So the only solution would be given a type, check whether it has some attributes. In this case, in the class attribute solution, I've to iterate all types under all assemblies to examine whether the attribute has been attached. That would cost like lots of waste, including a number of nuget trivia and infra packages. 

With the assembly attribute solution, I can filter in only one loop at assembly level (it's my view, not sure it's also the .NET's view : ) ) and retrieve the types from the assembly attributes list. I think this way is a little more scalable and extensible. The developer can also choose to manage the attribute all together in AssemblyInfo.cs or some other file served as a registry when the types grow more.

I borrowed some idea from https://github.com/Azure/azure-webjobs-sdk/blob/e5417775bcb8c8d3d53698932ca8e4e265eac66d/src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsStartupAttribute.cs Though the purpose not exactly same, but it also manage the types injection during bootstrapping time.

Hopefully, this explained but I'm opened to discuss more on this if there were anything not covered.